src/Compilers/CSharp/Portable/Symbols/Source/LambdaSymbol.cs

```diff
@@ -255,7 +256,20 @@ public override ImmutableArray<Location> Locations
         {
             get
             {
-                return ImmutableArray.Create<Location>(_syntax.Location);
```

Comment:
The way I read the issue, we can simply adjust the way the diagnostics is reported. I am not sure if we ever introduced a requirement for a location to be a single token. It would probably be better if the location was narrower, but this means that we would have to change behavior (potentially in a breaking way) of a public API.